extensions/advanced_ballistics/AdvancedBallistics.cpp

```diff
@@ -114,117 +115,56 @@ double calculateAtmosphericCorrection(double ballisticCoefficient, double temper
+        machNumbers = { 0.0, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30, 0.35, 0.40, 0.45, 0.50, 0.55, 0.60, 0.65, 0.70, 0.725, 0.75, 0.775, 0.80, 0.825, 0.85, 0.875, 0.90, 0.925, 0.95, 0.975, 1.0, 1.025, 1.05, 1.075, 1.10, 1.125, 1.15, 1.20, 1.25, 1.30, 1.35, 1.40, 1.50, 1.55, 1.60, 1.65, 1.70, 1.75, 1.80, 1.85, 1.90, 1.95, 2.00, 2.05, 2.10, 2.15, 2.20, 2.25, 2.30, 2.35, 2.40, 2.45, 2.50, 2.55, 2.60, 2.65, 2.70, 2.75, 2.80, 2.85, 2.90, 2.95, 3.00, 3.10, 3.20, 3.30, 3.40, 3.50, 3.60, 3.70, 3.80, 3.90, 4.00, 4.20, 4.40, 4.60, 4.80, 5.00 };
         break;
 
     case 8:
-        if (vel> 3571) { A = .0112263766252305; M = 1.33207346655961; }
-        else if (vel> 1841) { A = .0167252613732636; M = 1.28662041261785; }
-        else if (vel> 1120) { A = 2.20172456619625e-03; M = 1.55636358091189; }
-        else if (vel> 1088) { A = 2.0538037167098e-16; M = 5.80410776994789; }
-        else if (vel>  976) { A = 5.92182174254121e-12; M = 4.29275576134191; }
-        else if (vel>    0) { A = 4.3917343795117e-05; M = 1.99978116283334; }
+        dragCoefficients = { 0.2105, 0.2105, 0.2104, 0.2104, 0.2103, 0.2103, 0.2103, 0.2103, 0.2103, 0.2102, 0.2102, 0.2102, 0.2102, 0.2102, 0.2103, 0.2103, 0.2104, 0.2104, 0.2105, 0.2106, 0.2109, 0.2183, 0.2571, 0.3358, 0.4068, 0.4378, 0.4476, 0.4493, 0.4477, 0.4450, 0.4419, 0.4353, 0.4283, 0.4208, 0.4133, 0.4059, 0.3986, 0.3915, 0.3845, 0.3777, 0.3710, 0.3645, 0.3581, 0.3519, 0.3458, 0.3400, 0.3343, 0.3288, 0.3234, 0.3182, 0.3131, 0.3081, 0.3032, 0.2983, 0.2937, 0.2891, 0.2845, 0.2802, 0.2720, 0.2642, 0.2569, 0.2499, 0.2432, 0.2368, 0.2308, 0.2251, 0.2197, 0.2147, 0.2101, 0.2058, 0.2019, 0.1983, 0.1950, 0.1890, 0.1837, 0.1791, 0.1750, 0.1713 };
+        machNumbers = { 0.00, 0.05, 0.10, 0.15, 0.20, 0.25, 0.30, 0.35, 0.40, 0.45, 0.50, 0.55, 0.60, 0.65, 0.70, 0.75, 0.80, 0.825, 0.85, 0.875, 0.90, 0.925, 0.95, 0.975, 1.0, 1.025, 1.05, 1.075, 1.10, 1.125, 1.15, 1.20, 1.25, 1.30, 1.35, 1.40, 1.45, 1.50, 1.55, 1.60, 1.65, 1.70, 1.75, 1.80, 1.85, 1.90, 1.95, 2.00, 2.05, 2.10, 2.15, 2.20, 2.25, 2.30, 2.35, 2.40, 2.45, 2.50, 2.60, 2.70, 2.80, 2.90, 3.00, 3.10, 3.20, 3.30, 3.40, 3.50, 3.60, 3.70, 3.80, 3.90, 4.00, 4.20, 4.40, 4.60, 4.80, 5.00 };
         break;
 
     default:
         break;
 
     }
 
-    if (A != -1 && M != -1 && vel > 0 && vel < 10000) {
-        val = A * pow(vel, M) / DragCoefficient;
-        val = val / 3.2808399;
-        return val;
+    for (int i = 0; i < machNumbers.size(); i++) {
+        if (machNumbers[i] >= m) {
+            int previousIdx = std::max(0, i - 1);
+            double lcd = dragCoefficients[previousIdx];
```

Comment:
longer variable names would be neat here. I have no idea what lcd/lmn/cd mean.

Reply:
Agreed. Even though all we do here is a linear interpolation between the data points.